cmd/maya-apiserver/spc-watcher/storagepool_create.go

```diff
@@ -0,0 +1,181 @@
+
+func poolCreateWorker(pool *apis.CasPool) error {
+
+	glog.Infof("Creating storagepool for storagepoolclaim %s via CASTemplate", pool.StoragePoolClaim)
+
+	storagepoolOps, err := storagepool.NewCasPoolOperation(pool)
+	if err != nil {
+		return fmt.Errorf("NewCasPoolOperation failed error '%s'", err.Error())
+
+	}
+	_, err = storagepoolOps.Create()
+	if err != nil {
+		return fmt.Errorf("Failed to create cas template based storagepool: error '%s'", err.Error())
+
+	}
+
+	glog.Infof("Cas template based storagepool created successfully: name '%s'", pool.StoragePoolClaim)
+	return nil
+}
+
+// newCasPool will return a CasPool object
+func newCasPool(spcGot *apis.StoragePoolClaim, reSync bool, sparePoolCount int) (error, *apis.CasPool) {
+	// Validations for poolType
+	poolType := spcGot.Spec.PoolSpec.PoolType
+	if poolType == "" {
+		return errors.New("aborting storagepool create operation as no poolType is specified"), nil
+	}
+
+	if !(poolType == string(v1alpha1.PoolTypeStripedCPK) || poolType == string(v1alpha1.PoolTypeMirroredCPK)) {
+		return fmt.Errorf("aborting storagepool create operation as specified poolType is %s which is invalid", poolType), nil
+	}
+
+	diskType := spcGot.Spec.Type
+	if !(diskType == string(v1alpha1.TypeSparseCPK) || diskType == string(v1alpha1.TypeDiskCPK)) {
+		return fmt.Errorf("aborting storagepool create operation as specified type is %s which is invalid", diskType), nil
+	}
+	// The name of cas template should be provided as annotation in storagepoolclaim yaml
+	// so that it can be used.
+
+	// Check for cas template
+	casTemplateName := spcGot.Annotations[string(v1alpha1.SPCreateCASTemplateCK)]
+	if casTemplateName == "" {
+		return errors.New("aborting storagepool create operation as no cas template is specified"), nil
+	}
+	// Create an empty CasPool object and fill storagepoolcalim details
+	pool := &v1alpha1.CasPool{}
+	pool.StoragePoolClaim = spcGot.Name
+	pool.CasCreateTemplate = casTemplateName
+	pool.PoolType = spcGot.Spec.PoolSpec.PoolType
+	pool.MinPools = spcGot.Spec.MinPools
+	pool.MaxPools = spcGot.Spec.MaxPools
+	pool.Type = spcGot.Spec.Type
+	pool.ReSync = reSync
+	pool.SparePoolCount = sparePoolCount
+
+	// Fill the object with the disks list
+	pool.DiskList = spcGot.Spec.Disks.DiskList
+	// Check for disks
+	diskList := spcGot.Spec.Disks.DiskList
+	// If no disk are specified pool will be provisioned dynamically
+	if len(diskList) == 0 {
+		// newDisksList is the list of disks over which pool will be provisioned
+		err, newDisksList := getCasPoolDisk(pool)
+		if err != nil {
+			return err, nil
+		}
+		// Fill the object with the new disks list
+		pool.DiskList = newDisksList
+	}
+	return nil, pool
+}
+
+// getCasPoolDisk is a wrapper that will call getDiskList function to get the disk lists
+// that will be used to provision a storagepool dynamically
+
+func getCasPoolDisk(cp *apis.CasPool) (error, []string) {
+	// Performing valdations against CasPool fields
+	if cp.MaxPools < cp.MinPools {
```

Comment:
1) Add checks for integer values of MaxPools and MinPools.
2) MaxPools < MinPools check should ideally happen after all other checks.